pkg/config/config.go

```diff
@@ -250,13 +250,13 @@ func (c *ScrapeConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 	// Validate the scrape and timeout internal configuration. When /debug/pprof/profile scraping
 	// is enabled we need to make sure there is enough time to complete the scrape.
-	if c.ScrapeTimeout > c.ScrapeInterval {
-		return fmt.Errorf("scrape timeout must be smaller or equal to inverval for: %v", c.JobName)
-	}
-
 	if c.ScrapeTimeout == 0 {
-		c.ScrapeTimeout = c.ScrapeInterval
+		c.ScrapeTimeout = c.ScrapeInterval + model.Duration(3*time.Second)
 	}
+	if c.ScrapeTimeout <= c.ScrapeInterval {
+		return fmt.Errorf("scrape timeout must be greater than the interval: %v", c.JobName)
```

Comment:
hmm I think the timeout was required to be shorter to avoid potentially still being scraping the target when we start the next interval

It is the same logic in Prometheus: https://github.com/prometheus/prometheus/blob/v2.40.5/config/config.go#L293-L295

Reply:
An "infinitely" large timeout could be configured, resulting in scrapes pilling up on unresponsive targets until exhausting resources on the server

Reply:
Ah I see, nevermind, sounds about right, please carry on :slightly_smiling_face: